src/main/java/com/telefonica/iot/cygnus/backends/kafka/KafkaBackendImpl.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2016 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-cygnus (FI-WARE project).
+ *
+ * fiware-cygnus is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-cygnus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-cygnus. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with iot_support at tid dot es
+ */
+package com.telefonica.iot.cygnus.backends.kafka;
+
+import com.telefonica.iot.cygnus.log.CygnusLogger;
+import java.util.Properties;
+import kafka.admin.AdminUtils;
+import kafka.utils.ZKStringSerializer$;
+import org.I0Itec.zkclient.ZkClient;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+/**
+ *
+ * @author pcoello25
+ */
+
+public class KafkaBackendImpl implements KafkaBackend {
+    
+    private KafkaProducer<String, String> kafkaProducer;
+    private static final CygnusLogger LOGGER = new CygnusLogger(KafkaBackendImpl.class);
+    private ZkClient zookeeperClient;
+    private Properties properties;
+    
+    /**
+     * Constructor.
+     * @param brokerList
+     * @param zookeperEndpoint
+     */
+    public KafkaBackendImpl(String brokerList, String zookeperEndpoint) {
+        properties = new Properties();
+        properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+        properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        
+        // create the Zookeeper client
+        zookeeperClient = new ZkClient(zookeperEndpoint, 10000, 10000, ZKStringSerializer$.MODULE$);
```

Comment:
The advantage of having a permanent client is that no time is expent in establishing connections again and again. The disadvantage is the permanent connection may be closed... Is there any means of checking if a connection to Zookeeper is alive? If not, I would suggest to establish de connection each time a client is aimed to be used (despite the time expent in creating them).